### PR TITLE
Doc: revised README to be more helpful to casual GitHub visitors

### PR DESCRIPTION
**Pull request by andythenorth**
Think this is ready for nitpicking.

Note for reviewers: there's a lot of prior discussion about this on irc, anyone approaching it cold might find they're lacking the full tedious backstory

-  the goal is to make the readme more approachable and useful to GitHub casual visitors, without losing the ability to read it locally in the distributed bundle
- this may also end being visible in-game via #7786 but that is a separate concern 
- we're comfortable linking to remote pages that require an internet connection

For minor typos, it would be helpful to just fix them and force push.  Putting typo change requests on the GH PR review is pointless triple-handling for fixing minor typos.  Thx!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,183 +27,57 @@ OpenTTD is licensed under the GNU General Public License version 2.0,
...
+
+OpenTTD can be downloaded from the [official OpenTTD website](https://www.openttd.org/).
+
+Both 'stable' and 'nightly' versions are available for download.
+
+Most people will want the 'stable' version.
+
+The 'nightly' version includes the latest code changes and may be less reliable.
+
+On some platforms OpenTTD will also be available via your OS package manager or a similar service.
```
Comment (Eddi-z) [inline README.md:37]: this section reads a bit clumsy, i don't really like 1-line-paragraphs.

File: README.md [hunk 2]
```diff
@@ -35,183 +27,57 @@ OpenTTD is licensed under the GNU General Public License version 2.0,
...
-## 4.0) Installing and running OpenTTD
+### 1.3.1) Legacy support
+Platforms, languages and compilers change.
+We'll keep support going on old platforms as long as someone is interested in supporting them, except where it means the project can't move the project forward to keep up with language and compiler features.
 
-Installing OpenTTD is fairly straightforward. Either you have downloaded an
-archive which you have to extract to a directory where you want OpenTTD to
-be installed, or you have downloaded an installer, which will automatically
-extract OpenTTD in the given directory.
+We'll also do our best to maintain compatibility with old savegames, except where it becomes impractical.
```
Comment (Eddi-z) [inline README.md]: This statement is probably wrong, savegame compatibility is of much higher priority

File: README.md [hunk 3]
```diff
@@ -35,183 +27,57 @@ OpenTTD is licensed under the GNU General Public License version 2.0,
...
-currently being used. This means that if you use a config file in one of the
-shared directories, savegames will reside in the save/ directory next to the
-openttd.cfg file there.
+## 1.4) Installing and running OpenTTD
 
-If you want savegames and screenshots in the directory where the OpenTTD binary
-resides, simply have your config file in that location. But if you remove this
-config file, savegames will still be in this directory (see notes in
-section 4.2 'OpenTTD directories')
+OpenTTD is usually straightforward to install.  Installing varies by platform, and the wiki [includes an installation guide](https://wiki.openttd.org/Installation).
```
Comment (Eddi-z) [inline README.md]: Uhm, what?!?

File: README.md [hunk 4]
```diff
@@ -237,388 +104,96 @@ You need to copy the following files:
...
+### 1.5.2) Scenarios and height maps
+
+Scenarios and heightmaps can be added via the 'Check Online Content' button in the main menu.
+
+### 1.5.3) NewGRFs
 
-Under certain circumstance, especially on Ubuntu OpenTTD can be extremely slow
-and/or freeze. See known-bugs.txt for more information and how to solve this
-problem on your computer.
+A wide range of add-content is available as NewGRFs, including vehicles, industries, stations, landscape objects, town names and more.
```
Comment (Eddi-z) [inline README.md:137]: I think this can be elaborated a bit more

File: README.md [hunk 5]
```diff
@@ -237,388 +104,96 @@ You need to copy the following files:
...
-If the game is acting strange and you feel adventurous you can try the
-'`-d [[<name>=]<level>]`' flag, where the higher levels will give you more
-debugging output. The 'name' variable can help you to display only some type of
-debugging messages. This is mostly undocumented so best is to look in the
-source code file debug.c for the various debugging types. For more information
-look at https://wiki.openttd.org/Command_line.
 
-The most frequent problem is missing data files. Please install OpenGFX and
-possibly OpenSFX and OpenMSX. See section 4.1.1 for more information.
+## 1.5) Add-on content / mods
```
Comment (Eddi-z) [inline README.md:113]: some introductory paragraph in here along the lines of:
"Various methods to change the game content are available. Most of them can be installed with the ingame downloader. If you want to install them manually, refer to [it's complicated]"

Comment (andythenorth) [reply README.md:113]: Should we also reference patchpacks here?

Comment (glx22) [reply README.md:113]: Patchpacks are unofficial versions, so I think we are not supposed to mention them

File: README.md (hunk 2 above)
Comment (andythenorth) [reply README.md]: * this clause could be rewritten entirely
* or the 'except where it becomes impractical' could simply be dropped

I think that declaring 100% savegame compatibility as an absolute constraint isn't wise, and causes perception problems such as some of https://github.com/OpenTTD/OpenTTD/issues/7602  YMMV etc

File: README.md (hunk 3 above)
Comment (andythenorth) [reply README.md]: That's a very expressive and funny comment, but if this line is wrong or confusing, more specific reaction would be useful to improve it...

File: README.md (hunk 4 above)
Comment (andythenorth) [reply README.md:137]: A case could also be made for a 1 paragraph overview of all add-on content in OpenTTD, under the 'Add-on content / mods' heading, but I couldn't think what to write there.

File: README.md (hunk 5 above)
Comment (andythenorth) [reply README.md:113]: Patchpacks are mentioned in the official goals https://github.com/OpenTTD/OpenTTD/blob/master/CONTRIBUTING.md#i-do-not-agree-with-the-goals-of-the-official-branch-what-can-i-do-instead

BUT I think there's a difference between two messages:
* saying to developers 'we support forks and patchpacks" is fine
* saying to players "if you want to try different gameplay, download random unknown modified OpenTTD binaries" is not fine 

So I did leave patchpacks out of README.

File: README.md (hunk 1 above)
Comment (andythenorth) [reply README.md:37]: Me neither, but it's house style, and irc logs show it's been mentioned a few times as a requirement.

Comment (Eddi-z) [reply README.md:37]: No, you completely missed how that was meant.

it was meant that different sentences **in the same paragraph** should be on separate lines. not that each sentence should be a separate paragraph.

File: README.md (hunk 5 above)
Comment (Eddi-z) [reply README.md:113]: I think leaving out patchpacks here is fine.

File: README.md (hunk 4 above)
Comment (Eddi-z) [reply README.md]: "is usually straightforward to install." is tautological nonsense that has no place in a readme. people who have no trouble installing won't ever read this sentence (because why would you read a readme if you don't have a problem), and people who have problems installing will be mocked by this sentence.

File: README.md (hunk 3 above)
Comment (Eddi-z) [reply README.md]: "is usually straightforward to install." is tautological nonsense that has no place in a readme. people who have no trouble installing won't ever read this sentence (because why would you read a readme if you don't have a problem), and people who have problems installing will be mocked by this sentence.

Comment (andythenorth) [reply README.md]: Are the wiki installation guides sufficient for problem solving?  

Or to ask the question another way:
* option A is simply to reword this to address your point (which I agree with)
* option B is to identify that we need to provide better trouble-shooting guidance within the README (or linked from it)

Comment (Eddi-z) [reply README.md]: I'm not a fan of linking to the wiki. ideally, the readme should be somewhat self-contained.

File: README.md (hunk 1 above)
Comment (andythenorth) [reply README.md:37]: Reformatted most of the doc.  The argument about isolated sentences vs. paragraphs could go on ad-infinitum - some of the headings suffer from isolated sentences, but need to be written instructionally, with short clauses, visually distinct.

File: README.md [hunk 6]
```diff
@@ -35,183 +27,56 @@ OpenTTD is licensed under the GNU General Public License version 2.0,
...
-extract OpenTTD in the given directory.
+- Windows (Win32 GDI (faster) or SDL)
 
-OpenTTD looks in multiple locations to find the required data files (described
-in section 4.2). Installing any 3rd party files into a 'shared' location has
-the advantage that you only need to do this step once, rather than copying the
-data files into all OpenTTD versions you have.
+### 1.3.1) Legacy support
+Platforms, languages and compilers change.
+We'll keep support going on old platforms as long as someone is interested in supporting them, except where it means the project can't move the project forward to keep up with language and compiler features.
```
Comment (michicc) [inline README.md]: Nitpick: 'the project can't move the project'?

**Review by michicc — COMMENTED** [review]
👍

File: README.md (hunk 6 above)
Comment (andythenorth) [reply README.md]: Good spot, now resolved thanks

**Review by michicc — DISMISSED** [review]

File: README.md [hunk 7]
```diff
@@ -237,469 +103,115 @@ You need to copy the following files:
...
+
+The md5 implementation in src/3rdparty/md5 is licensed under the Zlib license.
+See the comments in the source files in src/3rdparty/md5 for the complete license text.
+
+The implementations of Posix getaddrinfo and getnameinfo for OS/2 in src/3rdparty/os2 are distributed partly under the GNU Lesser General Public License 2.1, and partly under the (3-clause) BSD license.
+The exact licensing terms can be found in src/3rdparty/os2/getaddrinfo.c resp. src/3rdparty/os2/getnameinfo.c.
+
+CONTRIBUTING.md is adapted from [Bootstrap](https://github.com/twbs/bootstrap/blob/master/CONTRIBUTING.md) under the [Creative Commons Attribution 3.0 Unported License](https://github.com/twbs/bootstrap/blob/master/docs/LICENSE) terms for Bootstrap documentation.
+
+## 4.0 Credits
```
Comment (LordAro) [inline README.md:211]: This section seems a bit pointless now that it's just a redirect. Remove?

File: README.md [hunk 8]
```diff
@@ -237,469 +103,115 @@ You need to copy the following files:
...
+This license applies to all files in this distribution, except as noted below.
+
+The squirrel implementation in src/3rdparty/squirrel is licensed under the Zlib license.
+See src/3rdparty/squirrel/COPYRIGHT for the complete license text.
+
+The md5 implementation in src/3rdparty/md5 is licensed under the Zlib license.
+See the comments in the source files in src/3rdparty/md5 for the complete license text.
+
+The implementations of Posix getaddrinfo and getnameinfo for OS/2 in src/3rdparty/os2 are distributed partly under the GNU Lesser General Public License 2.1, and partly under the (3-clause) BSD license.
+The exact licensing terms can be found in src/3rdparty/os2/getaddrinfo.c resp. src/3rdparty/os2/getnameinfo.c.
```
Comment (LordAro) [inline README.md]: I think some of these could do with some \`` wrapping them. definitely `getaddrinfo` & `getnameinfo`. Maybe all the paths as well?

File: README.md [hunk 9]
```diff
@@ -3,27 +3,19 @@
...
+    - 1.6) [OpenTTD directories](#16-openttd-directories)
+    - 1.7) [Compiling OpenTTD](#17-compiling-openttd)
+- 2.0) [Contact and community](#20-contact-and-community)
+    - 2.1) [Contributing to OpenTTD](#21-contributing-to-openttd)
+    - 2.2) [Reporting bugs](#22-reporting-bugs)
+    - 2.3) [Translating](#23-translating)
+- 3.0) [Licensing](#30-licensing)
+- 4.0) [Credits](#40-credits)
 
 ## 1.0) About
```
Comment (LordAro) [inline README.md:21]: "originally written by" (i can't comment on the line itself, grr, github). Section wants its line breaks... broken into the new style as well.

File: README.md [hunk 10]
```diff
@@ -35,183 +27,56 @@ OpenTTD is licensed under the GNU General Public License version 2.0,
...
-not be very difficult to port it to a new platform. The currently working
-platforms are:
+## 1.1) Downloading OpenTTD
+
+OpenTTD can be downloaded from the [official OpenTTD website](https://www.openttd.org/).
+
+Both 'stable' and 'nightly' versions are available for download:
+
+- most people should choose the 'stable' version, as this has been more extensively tested
+- the 'nightly' version includes the latest code changes and features, but may sometimes be less reliable
```
Comment (LordAro) [inline README.md]:
```suggestion
- the 'nightly' version includes the latest changes and features, but may sometimes be less reliable
```

File: README.md [hunk 11]
```diff
@@ -35,183 +27,56 @@ OpenTTD is licensed under the GNU General Public License version 2.0,
...
+For some platforms, you will need to refer to [the installation guide](https://wiki.openttd.org/Installation)
 
-Do NOT copy files included with OpenTTD into 'shared' directories (explained in
-the following sections) as sooner or later you will run into graphical glitches
-when using other versions of the game.
 
-#### 4.1.1) Free graphics and sound files
+### 1.4.1) Free graphics and sound files
 
 The free data files, split into OpenGFX for graphics, OpenSFX for sounds and
```
Comment (LordAro) [inline README.md:79]: Section might want a rework - very repetitive "Open*FX" and bare links

File: README.md [hunk 12]
```diff
@@ -237,469 +103,115 @@ You need to copy the following files:
...
-- Bug Reporters - Thanks for all bug reports
-- Chris Sawyer - For an amazing game!
+
+## 1.5) Add-on content / mods
+
+OpenTTD features multiple types of add-on content, which modify gameplay in different ways.
+
+Most types of add-on content can be downloaded within OpenTTD via the 'Check Online Content' button in the main menu.
+
+Add-on content can also be installed manually, but that's more complicated; the [OpenTTD wiki](https://wiki.openttd.org/OpenTTD) may offer help with that.
```
Comment (LordAro) [inline README.md]: Useless link is useless. At least should link to the NewGRF page

File: README.md [hunk 13]
```diff
@@ -237,469 +103,115 @@ You need to copy the following files:
...
+
+### 1.5.1) AI opponents
+
+OpenTTD comes without AI opponents, so if you want to play with AIs you have to download them.
+
+The easiest way is via the 'Check Online Content' button in the main menu.
+
+You can select some AIs that you think are compatible with your playing style.
+
+AI help and discussions may also be found in the [AI section of TT-Forums](https://www.tt-forums.net/viewforum.php?f=65).
```
Comment (LordAro) [inline README.md]:
```suggestion
AI help and discussions may also be found in the [AI section the Forum](https://www.tt-forums.net/viewforum.php?f=65).
```

File: README.md [hunk 14]
```diff
@@ -237,469 +103,115 @@ You need to copy the following files:
...
+
+NewGRFs can be added via the 'Check Online Content' button in the main menu.
+
+See also the wiki [guide to NewGRFs](https://wiki.openttd.org/NewGRF) and [TT-Forums graphics development section](https://www.tt-forums.net/viewforum.php?f=66).
+
+### 1.5.4) Game scripts
+
+Game scripts can provide additional challenges or changes to the standard OpenTTD gameplay, for example setting transport goals, or changing town growth behaviour.
+
+Game scripts can be added via the 'Check Online Content' button in the main menu.
```
Comment (LordAro) [inline README.md:147]: Very very repetitive "can be added via the ..." Put into parent section.

File: README.md (hunk 12 above)
Comment (LordAro) [reply README.md]: Maybe link to directory structure doc

File: README.md [hunk 15]
```diff
@@ -237,469 +103,115 @@ You need to copy the following files:
...
+
+
+## 2.0) Contact and Community
+
+'Official' channels
+
+- [OpenTTD website](https://www.openttd.org)
+- IRC chat using #openttd on irc.oftc.net [more info about our irc channel](https://wiki.openttd.org/Irc)
+- [OpenTTD on Github](https://github.com/openTTD/) for code repositories and for reporting issues
+- [forum.openttd.org](forum.openttd.org) - the primary community forum site for discussing OpenTTD and related games
```
Comment (LordAro) [inline README.md]: Link is broken

**Review by LordAro — CHANGES_REQUESTED** [review]

File: README.md (hunk 12 above)
Comment (andythenorth) [reply README.md]: Linking to the NewGRF page doesn't cover AIs, GS, scenarios, heightmaps....

I do think historically our explanation of add-on content has been a bit crap.  I don't have a great solution yet.

File: README.md (hunk 14 above)
Comment (andythenorth) [reply README.md:147]: It is very repetitive, but the information under each sub-heading has to stand alone as instructions.  Can't rely on referring up, or assuming the reader has read or retained earlier information.  

I tried to think of a better way, but this was it so far.

File: README.md (hunk 11 above)
Comment (andythenorth) [reply README.md:79]: +1 but no ideas for solutions...

**Review by LordAro — APPROVED** [review]
